txscript/sign/sign.go

```diff
@@ -118,15 +118,15 @@ func p2pkSignatureScript(tx *wire.MsgTx, idx int, subScript []byte,
 }
 
 // signMultiSig signs as many of the outputs in the provided multisig script as
-// possible. It returns the generated script and a boolean if the script
-// fulfills the contract (i.e. nrequired signatures are provided).  Since it is
-// arguably legal to not be able to sign any of the outputs, no error is
-// returned.
+// possible. It returns the generated script if 1 or more signature was provided,
+// otherwise it returns detailed error (while some parts of that error message
+// are expected artifacts of execution others might be useful when debugging
+// signing issues).
```

Comment:
> Since it is
// arguably legal to not be able to sign any of the outputs, no error is
// returned.

Pointing out this comment, I might be missing something, but I don't see why somebody would try to sign multisig transaction unless he expects at least 1 signature to succeed.